src/implementations/deploy/token-registry/token-registry.ts

```diff
@@ -1,38 +1,119 @@
   registryName,
   registrySymbol,
+  factoryAddress,
   network,
   gasPriceScale,
   dryRun,
   ...rest
-}: DeployTokenRegistryCommand): Promise<TransactionReceipt> => {
+}: DeployTokenRegistryCommand): Promise<{ contractAddress: string }> => {
   const wallet = await getWalletOrSigner({ network, ...rest });
-  const factory = new TradeTrustErc721Factory(wallet);
+  const chainId = await wallet.getChainId();
+  const deployContractAddress: DeployContractAddress = retrieveFactoryAddress(chainId, factoryAddress);
+
+  const factory = new ethers.Contract(
+    deployContractAddress.deployer,
+    TDocDeployer__factory.createInterface(),
+    wallet
+  ) as TDocDeployer;
+  signale.info(`Using ${deployContractAddress.titleEscrowFactory} as Title Escrow factory.`);
+
+  const initParam = encodeInitParams({
+    name: registryName,
+    symbol: registrySymbol,
+    deployer: await wallet.getAddress(),
+  });
+
   if (dryRun) {
-    const unsignedTx = factory.getDeployTransaction(registryName, registrySymbol, {});
-    const tx = await wallet.populateTransaction(unsignedTx);
-    await dryRunMode({
-      network,
-      gasPriceScale: gasPriceScale,
-      transaction: tx,
-    });
+    const estimatedGas: BigNumber = await factory.estimateGas.deploy(
+      deployContractAddress.tokenImplementation,
+      initParam
+    );
+    signale.info("Dry Run is unavailable for token reigstry deploy");
+    signale.info(`Estimated Gas Required: ${estimatedGas.toString()}`);
     process.exit(0);
   }
   const gasPrice = await wallet.provider.getGasPrice();
   signale.await(`Sending transaction to pool`);
-  const transaction = await factory.deploy(registryName, registrySymbol, { gasPrice: gasPrice.mul(gasPriceScale) });
-  trace(`Tx hash: ${transaction.deployTransaction.hash}`);
-  trace(`Block Number: ${transaction.deployTransaction.blockNumber}`);
-  signale.await(`Waiting for transaction ${transaction.deployTransaction.hash} to be mined`);
-  return transaction.deployTransaction.wait();
+
+  const transaction = await factory.deploy(deployContractAddress.tokenImplementation, initParam, {
+    gasPrice: gasPrice.mul(gasPriceScale),
+  });
+  trace(`Tx hash: ${transaction.hash}`);
+  trace(`Block Number: ${transaction.blockNumber}`);
+  signale.await(`Waiting for transaction ${transaction.hash} to be mined`);
+  const receipt = await transaction.wait();
+  const registryAddress = getEventFromReceipt<DeploymentEvent>(receipt, factory.interface.getEventTopic("Deployment"))
+    .args.deployed;
+  return { contractAddress: registryAddress };
+};
+
+interface Params {
+  name: string;
+  symbol: string;
+  deployer: string;
+}
+
+export const encodeInitParams = ({ name, symbol, deployer }: Params): string => {
+  return ethers.utils.defaultAbiCoder.encode(["string", "string", "address"], [name, symbol, deployer]);
+};
+
+const retrieveFactoryAddress = (chainId: number, factoryAddress: string | undefined): DeployContractAddress => {
```

Comment:
a test might be helpful here to read through whats to be expected and not from using this function